lua/weapons/weapon_peacekeeper.lua

```diff
@@ -0,0 +1,310 @@
+		vm:SendViewModelMatchingSequence(vm:SelectWeightedSequence( ACT_VM_PRIMARYATTACK ))
+
+		self.Owner:FireBullets( bullet )
+		self:EmitSound("Double_Barrel.Single")
+	
+		self:TakePrimaryAmmo(1)
+		self.Owner:ViewPunch( Angle( rnda,rndb,rnda ) )
+		self.Weapon:SetNextPrimaryFire( CurTime() + self.Primary.Delay )
+end
+
+function SWEP:SecondaryAttack()
+	--[[
+	local timerName = "ShotgunReload_" ..  self.Owner:UniqueID()
+	if (timer.Exists(timerName)) then return end
+
+	if self:Clip1()==0 then
+		self:Reload()
+		return
+	end
+
+	if ( !self:CanPrimaryAttack() ) then return end
+	local bullet = {}
+		bullet.Num = self.Primary.NumberofShots*2
+		bullet.Src = self.Owner:GetShootPos()
+		bullet.Dir = self.Owner:GetAimVector()
+		local ac = self:GetCone()
+		bullet.Spread = Vector(ac, ac, 0)
+		bullet.Tracer = 2
+		bullet.TracerName = "Tracer"
+		bullet.Force = self.Primary.Force
+		bullet.Damage = self.Primary.Damage
+		bullet.AmmoType = self.Primary.Ammo
+	local rnda = self.Primary.Recoil * -1 
+	local rndb = self.Primary.Recoil * math.random(-1, 1) 
+		local vm = self.Owner:GetViewModel()
+
+		--self.Owner:MuzzleFlash() -- Crappy muzzle light
+		self.Owner:SetAnimation( PLAYER_ATTACK1 )
+		vm:SendViewModelMatchingSequence(vm:SelectWeightedSequence( ACT_VM_PRIMARYATTACK ))
+		vm:SetPlaybackRate(BOLTACTIONSPEED)
+
+		self.Owner:FireBullets( bullet )
+		--self:EmitSound("dbarrel_dblast")
+	
+		self:TakePrimaryAmmo(2)
+		self.Owner:ViewPunch( Angle( rnda,rndb,rnda ) )
+		self.Weapon:SetNextPrimaryFire( CurTime() + self.Primary.Delay )
+		--self.Owner:SetAnimation( PLAYER_ATTACK1 )
+
+		
+		--timer.Simple( 2, function() self:SendWeaponAnim( ACT_VM_SECONDARYATTACK ) end )
+
+	--timer.Simple( self:SequenceDuration(), function() if ( !IsValid( self ) ) then return end self:SendWeaponAnim( ACT_VM_IDLE ) end )
+	--]]
+end
+
+function SWEP:Reload()
+
+	if not IsValid(self) then return end
+	if not IsValid(self.Owner) then return end
+	if not self.Owner:IsPlayer() then return end
+
+	local maxcap = self.Primary.ClipSize
+	local spaceavail = self.Weapon:Clip1()
+	local shellz = (maxcap) - (spaceavail) + 1
+
+	if (timer.Exists("ShotgunReload_" ..  self.Owner:UniqueID())) or (self.Owner.NextReload or 0) > CurTime() or maxcap == spaceavail then return end
+	
+	if self.Owner:IsPlayer() then
+		
+		if self.Owner:GetAmmoCount(self.Primary.Ammo) == 0 then return end
+
+		if self.Weapon:GetNextPrimaryFire() <= (CurTime()+2) then
+			self.Weapon:SetNextPrimaryFire(CurTime() + 2) -- wait TWO seconds before you can shoot again
+		end
+		self.Weapon:SendWeaponAnim(ACT_SHOTGUN_RELOAD_START) -- sending start reload anim
+		self.Owner:SetAnimation( PLAYER_RELOAD )
+		
+		self.Owner.NextReload = CurTime() + 1
+	
+		if (SERVER) then
+			self.Owner:SetFOV( 0, 0.15 )
+			--self:SetIronsights(false)
+		end
+	
+		if SERVER and self.Owner:Alive() then
+			local timerName = "ShotgunReload_" ..  self.Owner:UniqueID()
+			timer.Create(timerName,
+			(.5 + .05),
+			shellz,
```

Comment:
Isn't this a double barrel shotgun where the reload animation reloads both shells? Why is this coded like the type of shotgun you insert shells one at a time?